File: prototype/Maze3DAstraIntegrated/Assets/Assets/Scripts/utils/Position.cs
using System;
using UnityEngine;

namespace AssemblyCSharp{

	[Serializable]
	public class Position {

		public float x;
		public float y;
		public float z;

		public Position (){}

		public Position (Transform transform) {
			this.x = transform.position.x;
			this.y = transform.position.y;
			this.z = transform.position.z;
		}
	}
}

File: src/main/java/api/modules/utils/Position.java
package api.modules.utils;

import java.io.Serializable;

public class Position implements Serializable {

	private static final long serialVersionUID = 8450079507452997947L;
	
	public Double x;
	public Double y;
	public Double z;
	
	public Position () {}

	public Position (Position position) {
		this.x = position.getX();
		this.y = position.getY();
		this.z = position.getZ();
	}
	
	public Double getX() {
		return x;
	}

	public void setX(Double x) {
		this.x = x;
	}

	public Double getY() {
		return y;
	}

	public void setY(Double y) {
		this.y = y;
	}

	public Double getZ() {
		return z;
	}

	public void setZ(Double z) {
		this.z = z;
	}
	
	public String toString() {
		return "'position':{'x':" + getX() + ", 'y':" + getY() + ", 'z':" +getZ() + "}" ;
	}
}
